core/server/worker/src/main/java/alluxio/worker/block/RegisterStreamer.java

```diff
@@ -221,13 +221,13 @@ private void registerInternal() throws InterruptedException, DeadlineExceededExc
       iter++;
     }
 
-    // If the master side is closed before the client side, there is a problem
-    if (mFinishLatch.getCount() == 0) {
-      abort();
-    }
-
     // Wait for all batches have been ACK-ed by the master before completing the client side
     if (!mAckLatch.await(mResponseTimeoutMs * MAX_BATCHES_IN_FLIGHT, TimeUnit.MILLISECONDS)) {
+      // If the master side is closed before the client side, there is a problem
+      if (mFinishLatch.getCount() == 0) {
+        abort();
+      }
```

Comment:
@YangchenYe323 can you leave some comments on why this change is necessary?
@jiacheliu3 can you take a look here?

Reply:
Sure. The `mFinishLatch` check is moved after the `mAckLatch` check to catch more `InternalException` and `CancelledExeption` cases, not let them be masked as a `DeadlineExceededException`. Before the code looks like:
```java
if (mFinishLatch.getCount() == 0) {
        abort();
}
if (!mAckLatch.await(mResponseTimeoutMs * MAX_BATCHES_IN_FLIGHT, TimeUnit.MILLISECONDS)) {
...
throw DeadlineExceededException
```
It is easy for the server to raise an error **after** the finish check because not network latency, and this error will fail to be captured by the client.

Reply:
Makes sense, it is better to do the finish check after some waiting. Good catch!

In your example,
```
send all Worker requests
if (mFinishLatch.getCount() == 0) {
        abort();
}
if (!mAckLatch.await(mResponseTimeoutMs * MAX_BATCHES_IN_FLIGHT, TimeUnit.MILLISECONDS)) {
...
throw DeadlineExceededException
```
After all worker requests are sent, the worker should wait before expecting the responses from master, because the master needs time to process those requests.